lib/ProductOpener/Tags.pm

```diff
@@ -3539,6 +3541,39 @@ sub add_users_translations_to_taxonomy($) {
 
 }
 
+#Return the last occurence of the property if several are defined in ingredients.txt
+sub find_property_in_ingredient_tree($$$) {
+
+	my $tagtype = shift;
+	my $canon_tagid = shift;
+	my $property = shift;
+
+	while ($canon_tagid) 
+	{
+		if (exists $properties{$tagtype}{$canon_tagid})
+		{
+			foreach(keys $properties{$tagtype}{$canon_tagid})
+			{
+				if ($_ eq $property) 
+				{	
+					return "$properties{$tagtype}{$canon_tagid}{$_}\n";
+				}
+			}
+		}
+		if ($direct_parents{$tagtype}{$canon_tagid})
+		{
+			foreach(keys $direct_parents{$tagtype}{$canon_tagid}){$canon_tagid = $_;}
```

Comment:
This will assign the last parent to $canon_tagid

Reply:
yes, it will assign the canon_tagid of the parent to $canon_id and thanks to the while loop travel along the tree to the head, no ?

Reply:
ah ok sorry there was only one key during my test I didn't pay attention to the "S" in parents I thought it was only the +1 level, I will check the value of the key with a if, thanks

Reply:
I have tested with "my $lc = "fr"; my $tag = "flageolets verts extra-fins";" and the $direct_parents{$tagtype}{$canon_tagid} HASH contains only one direct parent and not both of them.  In which case it returns several values ?